File: solutions/7.reverse-integer/src/math.ts
/*
 * @lc app=leetcode.cn id=7 lang=typescript
 *
 * https://leetcode.cn/problems/reverse-integer/
 */

// @lc code=start
function reverse(x: number): number {
    let ret = 0;
    while (x !== 0) {
        if (ret < -Math.pow(2, 31) / 10 || ret > (Math.pow(2, 31) - 1) / 10) {
            return 0;
        }
        const digit = x % 10;
        x = Math.trunc(x / 10);
        ret = ret * 10 + digit;
    }
    return ret;
}
// @lc code=end

console.log(reverse(-123)); // 321

export { reverse };

File: solutions/7.reverse-integer/src/math.rs
/*
 * @lc app=leetcode.cn id=7 lang=rust
 *
 * https://leetcode.cn/problems/reverse-integer/
 */

pub struct Solution;

// @lc code=start
impl Solution {
    pub fn reverse(x: i32) -> i32 {
        let mut x = x;
        let mut ret = 0;
        while x != 0 {
            if ret < i32::MIN / 10 || ret > i32::MAX / 10 {
                return 0;
            }
            ret = ret * 10 + x % 10;
            x /= 10;
        }
        ret
    }
}
// @lc code=end
